docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -394,6 +396,52 @@ Some tools, such as `autoconf`, automatically determine what the compiler suppor
 
 ---
 
+### Warn on Unicode Unicode bidirectional (bidi) override characters |
+
+| [`-Wbidi-chars=any`](#-Wbidi-chars) | GCC | Warn on Unicode bidirectional (bidi) override characters |
+
+| Compiler Flag                                                                              | Supported since            | Description                                                                                  |
+| ------------------------------------------------------------------------------------------ | ----------------------- | -------------------------------------------------------------------------------------------- |
+| <span id="--Wbidi-chars">`-Wbidi-chars=any`</span>                                | GCC | Fortify sources with compile- and run-time checks for unsafe libc usage and buffer overflows         |
+
+#### Synopsis
+
+One kind of attack is "underhanded code", that is, maliciously misleading source code. Typically human review is one of the strongest defense against malicious code; underhanded code attempts to subvert this defense. For a general discussion, see [^Wheeler2020].
+
+"Trojan Source" is a specific kind of underhanded code that exploits the Bidirectional Algorithm in the Unicode Specification. Some natural languages (such as Arabic and Hebrew) are typically written right-to-left (RTL), while many others (such as English) are written left-to-right (LTR). Some texts must mix texts with different orders.  Unicode supports various control sequences to support this visual reordering. Unfortunately, attackers can use these control sequences so they can introduce vulnerabilities in source code that might not be visible to human reviewers.
+
+For more about Trojan Source see [trojansource.com](https://trojansource.codes/), which includes a copy of the paper "Trojan Source: Invisible Vulnerabilities" by Nicholas Boucher and Ross Anderson [^Boucher2021]. More context can be found [this Hacker News discussion](https://news.ycombinator.com/item?id=29062982) and [Wikipedia](https://en.wikipedia.org/wiki/Trojan_Source).  See, in particular, [CVE-2021-42574](https://nvd.nist.gov/vuln/detail/CVE-2021-42574) and [CVE-2021-42694](https://nvd.nist.gov/vuln/detail/CVE-2021-42684).
```

Comment:
Could this paragraph be turned into footnote and referred to also from the threat model? 

The references here would be relevant already there.

Reply:
They've been moved into the discussion about the Boucher paper.